### PR TITLE
16 add GitHub action for bun tests

### PR DESCRIPTION
**Pull request by makoeta**
Due to some bun errors, testing via GitHub Action is not possible with our code base.

Related issue: https://github.com/oven-sh/bun/issues/2927

The action is now just building the backend.

***Please squash the commits before merging.***

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
   "author": "Marvin Voss, Maximilian Koenig",
   "license": "ISC",
   "devDependencies": {
-    "@types/bun": "latest"
+    "@types/bun": "latest",
+    "bun-types": "^1.1.20"
```
Comment (hiqs-chris) [inline backend/package.json:16]: Why you installed bun-types additionally? The [NPM](https://www.npmjs.com/package/bun-types) page writes that these two are the same. @types/bun is an alias but the documentation writes to install that. So you could remove bun-types again?

File: .github/workflows/bunActions.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    branches: ["develop", "main", "**action**"]
+jobs:
+  build:
+    name: Build backend with bun
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: oven-sh/setup-bun@v2
+        with:
+          bun-version: 1.1.12
```
Comment (hiqs-chris) [inline .github/workflows/bunActions.yml:15]: Is that the correct version? It does not match the types-version?

**Review by hiqs-chris — CHANGES_REQUESTED** [review]
Check the packages

File: .github/workflows/bunActions.yml (hunk 2 above)
Comment (makoeta) [reply .github/workflows/bunActions.yml:15]: Yes, it is the version from the test system and the one I used for development.

File: backend/package.json (hunk 1 above)
Comment (makoeta) [reply backend/package.json:16]: This is a left over from testing on make the tests work on the gh pipe. So yes, can be removed.

**Pull request merged**